repos/system_upgrade/el7toel8/actors/satelliteupgradedatamigration/actor.py

```diff
@@ -0,0 +1,36 @@
+import glob
+import os
+
+from leapp.actors import Actor
+from leapp.libraries.stdlib import run
+from leapp.tags import IPUWorkflowTag, ApplicationsPhaseTag
+
+
+POSTGRESQL_DATA_PATH = '/var/lib/pgsql/data/'
+POSTGRESQL_SCL_DATA_PATH = '/var/opt/rh/rh-postgresql12/lib/pgsql/data/'
+
+
+class SatelliteUpgradeDataMigration(Actor):
+    """
+    No documentation has been provided for the satellite_upgrade_data_migration actor.
+    """
+
+    name = 'satellite_upgrade_data_migration'
+    consumes = ()
+    produces = ()
+    tags = (IPUWorkflowTag, ApplicationsPhaseTag)
+
+    def process(self):
+        # move PostgreSQL data to the new home
+        if os.path.exists(POSTGRESQL_DATA_PATH):
+            os.rmdir(POSTGRESQL_DATA_PATH)
+        if not os.path.exists('/var/lib/pgsql'):
+            os.mkdir('/var/lib/pgsql')
+        if os.path.exists(POSTGRESQL_SCL_DATA_PATH):
+            os.rename(POSTGRESQL_SCL_DATA_PATH, POSTGRESQL_DATA_PATH)
+        # fix back scenario files, might not be needed once we don't purge installer
+        for rpmsave_file in glob.glob('/etc/foreman-installer/scenarios.d/*.rpmsave'):
+            os.rename(rpmsave_file, rpmsave_file.replace('.rpmsave', ''))
+        # this hack is not needed after 6.10, as Foreman 3.1 has a fix in the installer for it
```

Comment:
6.10 is technically Foreman 2.5. Given 6.10 is already in beta, can we drop the workaround now?

Reply:
2.5 < 3.1? ;)
"after 6.10" means "with 7.0"